TestCases/compliance-level-3/0082-feel-coercion/0082-feel-coercion.dmn

```diff
@@ -157,8 +157,8 @@
     </businessKnowledgeModel>
 
     <businessKnowledgeModel name="bkm_005" id="_bkm_005">
-        <variable typeRef="number" name="bkm_005"/>
-        <encapsulatedLogic>
+        <variable name="bkm_005"/>
+        <encapsulatedLogic typeRef="number">
```

Comment:
Hi @opatrascoiu, in this case, the actual purpose of the test is to assert what an impl does in this exact circumstance - it is not an oversight.  The test description is "bkm type is non-list and bkm logic returns singleton list of conforming value - coercion list to val".  In essence, the logic returns an untyped value and that is coerced to the type of the bkm variable. It is not testing whether an impl does type inference, that is tested elsewhere.

To maintain the intention of the test, I believe the type should remain where it is.

Reply:
I think the description of the test should be ""bkm **return type/ output** type is non-list and bkm logic returns singleton list of conforming value - coercion list to val".

AFAIK variable.typeRef is the type of BKM (Any) -> List<number> encapsulated.typeRef is the return type List<number>.

I don't think moving the typeRef changes the intent of the test case.

Reply:
Hi @opatrascoiu. Merry Christmas to you.  If a variable may have a type, and that type may cause a coercion, then it should be the subject of a test case.  That is what this does.   I think maybe the gap is that we may not have a test that does a similar thing to the typeRef on the encapsulated logic.